transact/src/main/java/dev/dbos/transact/StartWorkflowOptions.java

```diff
@@ -27,31 +42,61 @@ public record StartWorkflowOptions(
 
   /** Construct with default options */
   public StartWorkflowOptions() {
-    this(null, null, null, null, null);
+    this(null, null, null, null, null, null);
   }
 
   /** Construct with a specified workflow ID */
   public StartWorkflowOptions(String workflowId) {
-    this(workflowId, null, null, null, null);
+    this(workflowId, null, null, null, null, null);
   }
 
   /** Produces a new StartWorkflowOptions that overrides the ID assigned to the started workflow */
   public StartWorkflowOptions withWorkflowId(String workflowId) {
     return new StartWorkflowOptions(
-        workflowId, this.timeout, this.queueName, this.deduplicationId, this.priority);
+        workflowId,
+        this.timeout,
+        this.queueName,
+        this.deduplicationId,
+        this.priority,
+        this.deadline);
   }
 
   /** Produces a new StartWorkflowOptions that overrides timeout value for the started workflow */
   public StartWorkflowOptions withTimeout(Timeout timeout) {
+    if (timeout != null && deadline != null) {
+      throw new IllegalArgumentException(
+          "should not specify a timeout if the deadline is already set");
+    }
     return new StartWorkflowOptions(
-        this.workflowId, timeout, this.queueName, this.deduplicationId, this.priority);
+        this.workflowId,
+        timeout,
+        this.queueName,
+        this.deduplicationId,
+        this.priority,
+        this.deadline);
+  }
+
+  /** Produces a new StartWorkflowOptions that overrides deadline value for the started workflow */
+  public StartWorkflowOptions withDeadline(Instant deadline) {
+    return new StartWorkflowOptions(
+        this.workflowId,
```

Comment:
We should be checking to ensure timeout and deadline aren't both set here as well. Actually, we should probably just it in the common validator (the code that checks to make sure we don't have a zero or negative timeout - not sure what Java calls that)

Reply:
lets open an issue to track